src/tokenizer.cpp

```diff
@@ -679,6 +682,10 @@ maybe_t<tok_t> tokenizer_t::next() {
     return result;
 }
 
+bool is_token_delimiter(wchar_t c, bool is_first, maybe_t<wchar_t> next) {
+    return c == L'(' || !tok_is_string_character(c, is_first, next);
```

Comment:
+bool is_token_delimiter(wchar_t c, bool is_first, maybe_t<wchar_t> next) {
+    return c == L'(' || !tok_is_string_character(c, is_first, next);

We also want closing parens, here is a test case:

    echo ()#)

Reply:
I see two different ways to interpret the test case you provided, and both yield the same result:
* expand empty list into an `echo` command, appending `#)` to each arg → a newline is printed
  - This is how I intended my code change to interpret this.
* expand empty list into an `echo` command, followed by a comment → a newline is printed
  - I assume this is how you meant we should interpret this.

Let's look at a different example:
```fish
echo (echo a)#b
```
I assume `a#b` should be printed in this case, since that is the current behavior. It also makes sense to me, since this interpretation is consistent with `echo a#b`. Do you agree?

Reply:
Sorry, you're right, # is not a comment after )
My bad -- using ()#) as example was foolish, since it is trivially empty because of cartesion product expansion